### PR TITLE
Working halofit

### LINKED ISSUE #3: The Nonlinear Matter Power Spectrum, $P^\mathrm{NL}_\mathrm{mm} (k, z)$

### LINKED ISSUE #4: The Linear Matter Power Spectrum, $P^\mathrm{L}_\mathrm{mm} (k, z)$

### PR DESCRIPTION
**Pull request by marcobonici**
This MR closes issues #3 and #4 .
The linear and nonlinear matter power spectrum have been implemented in the JAX Backend.
Both have been checked against the original jax-cosmo implementation.

### CODE REVIEW

File: cloelite/cosmology/jax_cosmology.py [hunk 1]
```diff
@@ -371,6 +371,35 @@ def int_sigma(logk):
...
+        \\sigma^2(R)= \\frac{1}{2 \\pi^2} \\int_0^\\infty \\frac{dk}{k} k^3 P(k,z) W^2(kR)
+
+        where
+
+        .. math::
+
+        W(kR) = \\frac{3j_1(kR)}{kR}
+        """
+        R = 8
+        kmin=0.0001
```
Comment (gcanasherrera) [inline cloelite/cosmology/jax_cosmology.py]: Passing these as args in the function...?

File: cloelite/cosmology/jax_cosmology.py [hunk 2]
```diff
@@ -371,6 +371,35 @@ def int_sigma(logk):
...
+        kmax=100.0
```
Comment (gcanasherrera) [inline cloelite/cosmology/jax_cosmology.py]: Same here

File: cloelite/cosmology/jax_cosmology.py [hunk 3]
```diff
@@ -371,6 +371,35 @@ def int_sigma(logk):
...
+
+        def int_sigma(logk):
+            k = np.exp(logk)
+            x = k * R
+            w = 3.0 * (np.sin(x) - x * np.cos(x)) / (x * x * x)
+            pk = self.transfer_Eisenstein_Hu(k) ** 2 * self.primordial_matter_power(k)
+            return k * (k * w) ** 2 * pk
+
+        #y = romb(int_sigma, np.log10(kmin), np.log10(kmax), divmax=7)
+        y = simps(int_sigma, np.log10(kmin), np.log10(kmax), N = 256)
```
Comment (gcanasherrera) [inline cloelite/cosmology/jax_cosmology.py:400]: Is N=256 enough?

**Review by gcanasherrera — COMMENTED** [review]
just a couple of things

File: cloelite/cosmology/jax_cosmology.py (hunk 3 above)
Comment (marcobonici) [reply cloelite/cosmology/jax_cosmology.py:400]: Yes, I checked with adaptive romberg. Difference is smaller than 0.01%.

**Pull request merged**